app/controllers/sessions_controller.rb

```diff
@@ -4,13 +4,12 @@ def new
 
   def create
     user = User.find_by(username: params[:session][:username])
-    if user && user.authenticate(params[:session][:password])
+    if user.admin? && user.authenticate(params[:session][:password])
```

Comment:
Woo, exorcism of the demonic nested if statements! This is way cleaner than it was before.